lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module

```diff
@@ -63,32 +59,27 @@ function dosomething_rogue_send_reportback_to_rogue($values, $user = NULL) {
 
   $run = dosomething_helpers_get_current_campaign_run_for_user($values['nid']);
 
-  $client = _dosomething_rogue_build_http_client();
+  $client = dosomething_rogue_client();
 
-  $options = [
-    'method' => 'POST',
-    'headers' => $client['headers'],
-    'data' =>
-        json_encode([
-          'northstar_id' => $northstar_id ? $northstar_id : NULL,
-          'drupal_id' => $user->uid,
-          'campaign_id' => $values['nid'],
-          'campaign_run_id' => isset($values['campaign_run_id']) ? $values['campaign_run_id'] : $run->nid,
-          'quantity' => $values['quantity'],
-          'why_participated' => $values['why_participated'],
-          'file' => $values['file'],
-          'file_id' => '', // @TODO - currently require by rogue, should not be.
-          'caption' => $values['caption'],
-          'status' => isset($values['status']) ? $values['status'] : 'pending',
-          'crop_x' => isset($values['crop_x']) ? $values['crop_x'] : NULL,
-          'crop_y' => isset($values['crop_y']) ? $values['crop_y'] : NULL,
-          'crop_width' => isset($values['crop_width']) ? $values['crop_width'] : NULL,
-          'crop_height' => isset($values['crop_height']) ? $values['crop_height'] : NULL,
-          'crop_rotate' => isset($values['crop_rotate']) ? $values['crop_rotate'] : NULL,
-        ]),
+  $data = [
+        'northstar_id' => $northstar_id ? $northstar_id : NULL,
+        'drupal_id' => $user->uid,
+        'campaign_id' => $values['nid'],
+        'campaign_run_id' => $run->nid,
+        'quantity' => $values['quantity'],
+        'why_participated' => $values['why_participated'],
+        'file' => $values['file'],
+        'file_id' => '', // @TODO - currently require by rogue, should not be.
+        'caption' => $values['caption'],
+        'status' => isset($values['status']) ? $values['status'] : 'pending',
+        'crop_x' => $values['crop_x'],
+        'crop_y' => $values['crop_y'],
+        'crop_width' => $values['crop_width'],
+        'crop_height' => $values['crop_height'],
+        'crop_rotate' => $values['crop_rotate'],
   ];
 
-  $response = drupal_http_request($client['base_url'] . '/reportbacks', $options);
+  $response = $client->post($client->getBaseUri() . '/reportbacks', $data);
```

Comment:
It will still work if you include it each time, but providing a relative URL should automatically prepend the base URL specified when you created the client above (in theory! 😆)